### PR TITLE
add-language functionality

### PR DESCRIPTION
**Pull request by ivanhofer**

### CODE REVIEW

**Comment by NilsJacobsen** [comment]
Thank you @ivanhofer, gonna pick it up and add the add string modal.

File: source-code/website/src/pages/editor/@host/@owner/@repository/State.tsx [hunk 1]
```diff
@@ -455,8 +478,7 @@ export function EditorStateProvider(props: { children: JSXElement }) {
 		if (!inlangConfig.error && inlangConfig() && fsChange()) {
 			// setting the origin store because this should not trigger
```
Comment (ivanhofer) [inline source-code/website/src/pages/editor/@host/@owner/@repository/State.tsx:479]: We have a blocker.
We set the resources with `setOriginResources` in line 455, then write everything to disk, with `writeResources` which at the end updates `fsChange`. This triggers this code and calls `readResources`. But read resources will only read the old files from the file system. Then newly written resource does not get loaded.

Comment (ivanhofer) [reply source-code/website/src/pages/editor/@host/@owner/@repository/State.tsx:479]: fixed it!

**Comment by ivanhofer** [comment]
This PR is ready for review!

**Comment by NilsJacobsen** [comment]
@ivanhofer i merged and tested it locally.
But it doesn't pass the tests in github actions.

**Comment by NilsJacobsen** [comment]
@ivanhofer did you already hear about that octokit/core error?

**Comment by ivanhofer** [comment]
@NilsJacobsen I have pushed some commits. Please check if it works

**Comment by NilsJacobsen** [comment]
There is only a lint error left. If you don't fix it, i'm gonna do it tomorrow:)

**Pull request merged**